src/Exchange/Rate.php

```diff
@@ -81,6 +81,11 @@ public function convert(Money $money): Money
         return $money->convert($this->currencyPair->getQuoteCurrency(), $this->ratio, PHP_ROUND_HALF_UP);
     }
 
+    public function withRatio(float $ratio): self
```

Comment:
I don't understand the purpose of this method.

Reply:
For example, why not a method `withMarkup(float $ratio)`... or similar?

Reply:
It's factory method for rate with modified ratio. For example, on DCC

```php
$rate = $rate->withRatio($rate->getRatio() * $markup);
```

Reply:
> why not a method `withMarkup`

This is only one use case. In DCC we have two cases of modification exchange rate:

1) Rate with markup, a.k.a final rate for user
2) Markup rate - to calculate revenue

I think more generic method would be useful anyway.

Reply:
Even this seems wrong though:

```
$markupRate = $rate->withRatio($rate->getRatio() * $markup);
```

Markup is usually not written as "ready-to-multiply"... so for example, it would be 

```
$markupRate = $rate->withRatio($rate->getRatio() * (1 + $markup/100));
```

I don't understand what you mean about point #2 and I do not think it is a feasible use case, because money must be allocated to the minor unit (in our case for markup charges cents in USD).